cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,363 @@
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type from supported repo types:" + supportedTypesMarkdown(repositoryTypes()),
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			NetworkAddressKey: {
+				Description:  "The network address that the sidecar listens on.",
+				Type:         schema.TypeSet,
+				Optional:     true, // This field is required, due to ExactlyOneOf directive
+				ExactlyOneOf: []string{NetworkAddressKey},
+				MaxItems:     1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						HostKey: {
+							Description: "Host where the sidecar will listen for the given repository. Omit to listen on all interfaces.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						PortKey: {
+							Description: "Port where the sidecar will listen for the given repository.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MySQLSettingsKey: {
+				Description: "MySQL settings represents the listener settings for a [`mysql`, `galera`, `mariadb`] data repository.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						DbVersionKey: {
+							Description: "MySQL DB version. Required (and only relevant) for listeners of type `mysql`.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						CharacterSetKey: {
+							Description: "MySQL character set. Optional and only relevant for listeners of type `mysql`.",
```

Comment:
Nit: Just to keep consistency in the docs.
```suggestion
							Description: "MySQL character set. Optional (and only relevant) for listeners of type `mysql`.",
```